### DISCUSSION TITLE
제페토 멀티플레이 동기화에 대해서

### DISCUSSION THREAD

**Discussion by pda017** [naverz/zepeto-studio-kor#233] [5. Q&A]
멀티플레이시 동기화 하는방법이 두가지 있는거 같은데요.
Room State로 데이터를 동기화 시키는 방법이 있고
Send로 서버에 메세지를 보낸후 Broadcast로 동기화 시키는 방법이 있는거 같습니다.
Room State가 변경된 데이터만 새로 갱신시키는 방식인가요 아니면 모든 데이터를 다시 전송하는 방식인가요?
만약에 Room State가 변경된 데이터만 갱신된다면 Room State로만 동기화 시키는게 적절한가요?
Send Broadcast에 비해 코스트가 심하게 큰지 아닌지가 궁금하네요.

**Comment by hyeonjin-jo** [comment]
안녕하세요.

Room State의경우 변경된 데이터만 갱신시키는 방식이 맞습니다.

클라이언트에서는 OnStateChange로 전체를 받는 것으로 보이나 내부적으로는 변경된 데이터만 보내고 있습니다.

기본적으로 State를 활용한 동기화를 권장 드리고 특정 이벤트에서 Broadcast를 적절하게 사용하는 방식이 좋을 것 같습니다.

**Reply by pda017** [reply]
공식 가이드에 그점을 알려주시면 좋을거 같습니다 ^^
답변 감사합니다.

**Reply by Sonak0930** [reply]
안녕하세요. 동기화 관련해서 한 가지만 여쭙고 싶습니다. 공식 가이드를 따라서 하고 있는데, ClientStarter라는 스크립트가 있습니다. 여기서 SendTransform이라는 룸 데이터를 생성한 뒤에 룸에 업데이트를 하는 방식이 있습니다. 혹시 동기화 하실 때 어떤 함수를 부르시는지 알 수 있을까요?  이 함수만 직접 호출한다고 해서 동기화가 바로 되지는 않는 거 같았습니다. 비슷한 스레드를 찾고있는데, 혹시 알고 계시면 공유 부탁드립니다 ㅠㅠ.

`public SendTransform(transform:Transform)
    {
        const data = new RoomData();
        const pos = new RoomData();

        pos.Add("x",transform.position.x);
        pos.Add("y",transform.position.y);
        pos.Add("z",transform.position.z);
        data.Add("position",pos.GetObject());

        const rot = new RoomData();
        rot.Add("x",transform.eulerAngles.x);
        rot.Add("y",transform.eulerAngles.y);
        rot.Add("z",transform.eulerAngles.z);
        data.Add("rotation",rot.GetObject());

        this.room.Send("onChangedTransform",data.GetObject());

    }
`
